src/fastoad/openmdao/variables/variable_list.py

```diff
@@ -75,6 +76,14 @@ class VariableList(list):
         print( 'var/2' in vars_A.names() )
     """
 
+    # We override __eq__, so we must explicitly set __hash__ = None.
+    # This makes the class unhashable, which is the correct behavior for mutable types.
+    # It prevents inconsistent behavior when using VariableList in sets or as dict keys.
+    # See: https://docs.astral.sh/ruff/rules/eq-without-hash/
+    # Even though we inherit from list (which is unhashable),
+    # overriding __eq__ disables the inherited __hash__.
+    __hash__ = None
```

Comment:
This is a fix right ? Does it have anything to do with the new rules you implemented ?